BUG: Test workflows not failing on notebook exceptions

I think we need to revisit the testing script to ensure that it's working properly. I pushed up a [dummy commit](https://github.com/scientific-python/executable-tutorials/pull/50/commits/8332c7938d90c6c578eb6b5190a1a5ddf9bc496d) to ensure that CI would fail on notebook failure. The sphinx-based builds failed, but the test workflows all passed!

It's a return status issue, and thus I think a bug in the shell script, pytest actually has a failing status:

```
 ---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[1], line 1
----> 1 foo

NameError: name 'foo' is not defined

============================= slowest 10 durations =============================
3.25s call     tutorials/matplotlib/interactive_mpl.ipynb::Cell 0
1.31s setup    tutorials/executable/basics.ipynb::Cell 0
0.83s setup    tutorials/matplotlib/static_mpl.ipynb::Cell 0
0.71s setup    tutorials/matplotlib/interactive_mpl.ipynb::Cell 0
0.33s call     tutorials/matplotlib/static_mpl.ipynb::Cell 0
0.15s call     tutorials/matplotlib/static_mpl.ipynb::Cell 1
0.15s call     tutorials/executable/basics.ipynb::Cell 3
0.14s call     tutorials/matplotlib/static_mpl.ipynb::Cell 2
0.11s teardown tutorials/matplotlib/interactive_mpl.ipynb::Cell 4
0.11s teardown tutorials/executable/basics.ipynb::Cell 4
=========================== short test summary info ============================
FAILED tutorials/executable/basics.ipynb::Cell 4
========================= 1 failed, 12 passed in 7.57s =========================
```

--- a/pixi.toml
+++ b/pixi.toml
@@ -46,6 +46,7 @@ myst-nb = ">=1.1"
 jupytext = ">=1.16"
 sphinx-book-theme = ">=1.1"
 sphinx-copybutton = ">=0.5"
+pytest-custom_exit_code = "*"
 
 [feature.py312.dependencies]
 python = "3.12.*"

--- a/test.sh
+++ b/test.sh
@@ -29,6 +29,12 @@ done
 
 pytest --nbval-lax -vv --suppress-no-test-exit-code --durations=10
 
+_exitval="$?"
+
+if [ $_exitval > 0 ]; then
+    exit $_exitval
+fi
+
 # Clean up ipynb files that were converted.  Any stray ipynb files that were
 # _not_ the result of conversion from Markdown will be left alone.
 for file in "${notebook_files[@]}"; do

